<launch>


    <arg name="base_name" default="buxx"/>
    <arg name="machine" default="localhost"/>
    <arg name="queue_size"        default="5"/>
    <arg name="nodelet_manager" default="$(arg base_name)"/>
    <arg name="start_manager" default="true"/>
    <arg name="use_machine" default="true"/>
    <arg name="respawn" default="true"/>
    <arg name="use_nodelet" default="true"/>
    <arg name="output" default="screen"/>

    <machine name="localhost" address="localhost" if="$(arg use_machine)"/>

    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager"
          if="$(arg start_manager)" machine="$(arg machine)" output="screen"/>

    <!-- sd point cloud (512 x 424) -->
    <node pkg="nodelet" type="nodelet" name="$(arg base_name)_points_xyzrgb_sd"
          args="load depth_image_proc/point_cloud_xyzrgb $(arg nodelet_manager)" >
        <remap from="rgb/camera_info" to="$(arg base_name)/sd/camera_info"/>
        <remap from="rgb/image_rect_color" to="$(arg base_name)/sd/image_color_rect"/>
        <remap from="depth_registered/image_rect" to="$(arg base_name)/sd/image_depth_rect"/>
        <remap from="depth_registered/points" to="$(arg base_name)/sd/points"/>
        <param name="queue_size" type="int" value="$(arg queue_size)"/>
    </node>

    <!-- cloud rotation -->
    <node pkg="nodelet" type="nodelet" name="$(arg base_name)_cloud_rotation"
          args="load cloud_rotation $(arg nodelet_manager)" >
        <remap from="$(arg base_name)_cloud_rotation/cloud_in"                    to="$(arg base_name)/sd/points"/>
        <remap from="$(arg base_name)_cloud_rotation/imu_data"                    to="$(arg base_name)/imu_reader/imu_data"/>
        <param name="queue_size" type="int" value="$(arg queue_size)"/>
    </node>

    <!--plane segmentation-->
    <node pkg="nodelet" type="nodelet" name="$(arg base_name)_peac" machine="$(arg machine)"
          args="load peac/seg $(arg nodelet_manager)" respawn="$(arg respawn)">
        <param name="show_seg_viewer" type="bool" value="false" />
        <remap from="$(arg base_name)_peac/cloud_in" to="$(arg base_name)_cloud_rotation/cloud_xyzrgba"/>
    </node>

    <!--plane segmentation node version-->
    <!--<node pkg="peac" type="peac_ros_node" name="$(arg base_name)_plane_seg" machine="$(arg machine)" output="screen">-->
    <!--<remap from="$(arg base_name)_plane_seg/cloud_in"   to="$(arg base_name)_cloud_rotation/cloud_xyzrgba"/>-->
    <!--</node>-->

    <!-- segmentation result -->
    <node pkg="image_view" type="image_view" name="image_view" machine="$(arg machine)"
          args="image:=$(arg base_name)_peac/seg_image"/>

    <!-- stair_modeling -->
    <!-- <node pkg="nodelet" type="nodelet" name="$(arg base_name)_stair_modeling" machine="$(arg machine)"
          args="load peac/stair_modeling $(arg nodelet_manager)" respawn="$(arg respawn)">
        <remap from="$(arg base_name)_stair_modeling/vec_planes" to="$(arg base_name)_peac/vec_planes"/>
    </node> -->

    <!--<node pkg="stair_modeling" type="stair_modeling_node" name="$(arg base_name)_stair_modeling" machine="$(arg machine)" output="screen">-->
    <!--<remap from="peac_plane_seg/vec_planes" to="$(arg base_name)_peac/vec_planes"/>-->
    <!--</node>-->

    <!-- rqt_reconfigure -->
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="param_setting" machine="$(arg machine)"/>

</launch>
